cdm_souffleur/services/authorization_service.py

```diff
@@ -1,21 +1,40 @@
-import random
-import string
-
+from werkzeug.utils import redirect
 from cdm_souffleur.model.unauthorized_reset_pwd_request import unauthorized_reset_pwd_request
 from cdm_souffleur.model.user import *
 from cdm_souffleur.services.mailout_service import send_email
 from cdm_souffleur.utils import InvalidUsage
+from cdm_souffleur.utils.constants import REGISTRATION_LINK_EXPIRATION_TIME, PASSWORD_LINK_EXPIRATION_TIME
 from cdm_souffleur.utils.exceptions import AuthorizationError
 from cdm_souffleur import bcrypt
+from cryptography.fernet import Fernet
+import atexit
+from apscheduler.schedulers.background import BackgroundScheduler
+
 
-unconfirmed_users = {}
 user_registration_links = {}
 reset_pwd_links = {}
+key = Fernet.generate_key()
+fernet = Fernet(app.config.get('EMAIL_ENCODE_KEY'))
+
+
+def refresh_registration_links():
+    global user_registration_links, reset_pwd_links
+    user_registration_links = {key: value for key, value in user_registration_links.items() if
+      (datetime.datetime.now() - value['date_time']).total_seconds() < REGISTRATION_LINK_EXPIRATION_TIME}
+    reset_pwd_links = {key: value for key, value in reset_pwd_links.items() if
+      (datetime.datetime.now() - value).total_seconds() < PASSWORD_LINK_EXPIRATION_TIME}
+
+
+scheduler = BackgroundScheduler()
+scheduler.add_job(func=refresh_registration_links, trigger="interval", seconds=3600)
```

Comment:
interval can be 24 hours and please add a comment with time in a readable format